common/tools/dev-tool/src/util/testProxyUtils.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { spawn } from "child_process";
+import path from "path";
+import { IncomingMessage, request, RequestOptions } from "http";
+import fs from "fs-extra";
+import { createPrinter } from "./printer";
+
+const log = createPrinter("test-proxy");
+export async function startProxyTool(mode: string | undefined) {
+  log.info(`===TEST_MODE="${mode}"===`);
+  log.info(
+    `Attempting to start test proxy at http://localhost:5000 & https://localhost:5001.\n`
+  );
+
+  const subprocess = spawn(await getDockerRunCommand(), [], {
+    shell: true
+  });
+  
+  const outFileName = "test-proxy-output.log";
+  const out = fs.createWriteStream(`./${outFileName}`,{flags:'a'});
+  subprocess.stdout.pipe(out);
+  subprocess.stderr.pipe(out);
+  
+  log.info(`Check the output file "${outFileName}" for test-proxy logs.`);
+}
+
+async function getRootLocation(start?: string): Promise<string> {
+  start ??= process.cwd();
+  if (await fs.pathExists(path.join(start, "rush.json"))) {
+    return start;
+  } else {
+    const nextPath = path.resolve(start, "..");
+    if (nextPath === start) {
+      throw new Error("Reached filesystem root, but no rush.json was found.");
+    } else {
+      return getRootLocation(nextPath);
+    }
+  }
+}
+
+async function getDockerRunCommand() {
+  const repoRoot = await getRootLocation(); // /workspaces/azure-sdk-for-js/
+  const testProxyRecordingsLocation = "/etc/testproxy";
```

Comment:
nit: `/etc` is usually for system configurations. maybe use another directory?

Reply:
@scbedd

Reply:
https://github.com/Azure/azure-sdk-tools/issues/2279

@HarshaNalluru the PR is submitted to sync the proxy tool tag update, I will need to change this script as well. That will keep everything in sync.